### DISCUSSION TITLE
4.0" ST7796S + FT6336U + ESP32-S3-Devkitc-1 N8R2 Problem

### DISCUSSION THREAD

**Discussion by Mahdiahmadimehr** [Bodmer/TFT_eSPI#3485] [Compatible displays and setup files]
Hi Everyone
i'm New to TFT LCD's And I'm Trying To Use http://www.lcdwiki.com/4.0inch_Capacitive_SPI_Module_ST7796 as HMI For my Device.
So I need Help I Used TFT_eSPI And LVGL but none of them works.
i just can send simple Fill Color to LCD using SPI Commands
i Tried ESP-IDF And Arduino IDE none of them works correct
so here is my prooblem with Arduino IDE
Everytime i use TFT_eSPI it Compile And Upload to board but everytime my ESP Boot On Download Mode:
![Screenshot 2024-09-20 194506](https://github.com/user-attachments/assets/1d25e375-743a-45ad-a5ca-c4e6b802789f)
I Tried 2 different Config File For User_Setup.h And different SPI Pins Like This

Config 1:
#define TFT_MISO 5
#define TFT_MOSI 6
#define TFT_SCLK 4
#define TFT_CS   7   // Chip select control pin
#define TFT_DC   9   // Data Command control pin
#define TFT_RST  10   // Reset pin (could connect to RST pin)

Config 2:
#define TFT_MISO 12
#define TFT_MOSI 13
#define TFT_SCLK 14
#define TFT_CS   11   // Chip select control pin
#define TFT_DC   17   // Data Command control pin
#define TFT_RST  18   // Reset pin (could connect to RST pin)

But Still Not Working!
it's important to me to make it works

**Comment by elgerg** [comment]
What speed did you set your serial output to as it looks like your output is garbled.

Also, it looks like your board uses Quad SPI for flash and you have set Dual SPI (DIO).

https://docs.espressif.com/projects/esp-idf/en/stable/esp32s3/hw-reference/esp32s3/user-guide-devkitc-1.html#ordering-information

Maybe you should get a blink sketch with serial output to work before trying to go further?

**Reply by Mahdiahmadimehr** [reply]
default Speed is 115200 And Every Other Sketchs Work Well
I Tried Different ESP-IDF Project Like WIFI Scan, WebServer, Blink And ...
All of hthem Works Well 
![com-03-090-4](https://github.com/user-attachments/assets/2a0cd08d-2c93-4b84-8462-2477955ec5aa)

**Comment by elgerg** [comment]
What version of TFT_eSPI and Arduino package are you using?
Could your problem be the same as:
https://github.com/Bodmer/TFT_eSPI/issues/3329
?

**Reply by Mahdiahmadimehr** [reply]
I Use 2.5.43

![image](https://github.com/user-attachments/assets/8400963e-bdd2-484d-bc53-1154f1caa07a)

**Comment by elgerg** [comment]
What version of Arduino package are you using?
Look in board manager

**Reply by Mahdiahmadimehr** [reply]
Arduino AVR Boards V1.8.6 And ESP32 V3.0.5
I Used ESP-IDF And Build TFT_eSPI But Still Same Problem
When I Use it ESP32 Stuck On Download Mode
I Tried SPI On 36-39 And SPI On 9-14

**Comment by elgerg** [comment]
Did you see the comment in the issue I mentioned?
Espressif Arduino core 2.0.14 works, try downgrading to see if you are having the same problem or if it fixes it. If it fixes it then you could try upgrading back to 3.0.5 and applying the fixes as mentioned in the issue.

**Reply by Mahdiahmadimehr** [reply]
Yes i'm on that please give me a second i will be back

**Reply by Mahdiahmadimehr** [reply]
My Friend Thank You So Much
i Used #define USE_HSPI_PORT Which means SPI2 on ESP32-S3 And Downgrade Arduino ESP32 to 2.0.13 
rewiring TFT to My Board and TFT_eSPI works Fine but Still lvgl not working

**Reply by elgerg** [reply]
2.0.14 should work fine. Some have had success upgrading to 3.x but have had to make edits to come core files.

With LVGL it might be best to create a topic on their forum:
https://forum.lvgl.io/c/how-to/6

If I see something there I will also try to help if I can as I also use LVGL, not been brave enough to go to version 9 yet though...

Might be worth closing this thread down as it'll just cause more confusion with the breaking change between 2.0.14 and 2.0.15/3.x

Regards

**Reply by Mahdiahmadimehr** [reply]
i will do that thank you so much again ❤️
